Products/RhaptosModuleEditor/skins/rhaptos_module_editor/publishContent.cpy

```diff
@@ -39,6 +39,37 @@ if hasattr(context,'excludedIds'):
     if excludedexists:
         context.manage_delObjects(excludedexists)
 
+# Generate SVGs for MathML
+import requests
+from lxml import etree
+# FIXME Hardcoded URL
+url = "http://localhost:5689/"
+
+file = context.getDefaultFile()
+xml = etree.fromstring(file.getSource())
+mathml_namespace = "http://www.w3.org/1998/Math/MathML"
+mathml_blocks = xml.xpath(
+    '//m:math[not(/m:annotation-xml[@encoding="image/svg+xml"])]',
+    namespaces={'m': mathml_namespace})
+for mathml_block in mathml_blocks:
+    # Submit the MathML block to the SVG generation service.
+    payload = {'MathML': etree.tostring(mathml_block)}
+    response = requests.post(url, data=payload)
+    # Inject the SVG into the MathML as an annotation
+    # only if the resposne was good, otherwise skip over it.
+    semantic_block = mathml_block.getchildren()[0]
```

Comment:
This assumes the presence of a m:semantics tag as first child of the m:math tag, since the annotation-xml is only valid in such a context. Unfortunately, only some of our math has such semantics tags: (mostly things imported from Word). We'll need to detect that case, and wrap the existing content is such a tag first if it's not there